test_reporting/report_data_storage.py

```diff
@@ -172,6 +195,13 @@ def upload_report(self, report_json: Dict, external_tracking_id: str = "", repor
                 This id does not have to be unique.
             report_guid: A randomly generated UUID that is used to query for a specific test run across tables.
         """
+        os_version = self._parse_os_version(image_url)
+        if not report_json:
+            print("Uploaded file is not found or empty. We only upload pipeline results and summary")
+            self._upload_pipeline_results(external_tracking_id, report_guid, testbed, os_version)
```

Comment:
Sometimes we may use this tool to upload test results manually. If running this script always need to get pipeline results,  then we can't use this script to manually upload test results. We need to make the step of getting pipeline results optional.

Reply:
@wangxin Add a new `collect_azp_results.py` to collect task status for specific pipeline.